modules/exploits/windows/ftp/wing_ftp_admin_exec.rb

```diff
@@ -48,40 +53,53 @@ def initialize(info = {})
   end
 
   def check
-    res = send_request_cgi(
-      {
-        'uri'     =>  '/admin_login.html',
-        'method'  => 'GET'
-      })
+    @session_cookie = authenticate(datastore['USERNAME'], datastore['PASSWORD'])
 
-    if !res
-      fail_with(Failure::Unreachable, "#{peer} - Admin login page was unreachable.")
-    elsif res.code != 200
-      fail_with(Failure::NotFound, "#{peer} - Admin login page was not found.")
-    elsif res.body =~ /Wing FTP Server Administrator/ && res.body =~ /2003-2014 <b>wftpserver.com<\/b>/
-      return Exploit::CheckCode::Appears
-    end
+    ver = send_request_cgi(
+      'uri'       => '/admin_license.html',
+      'method'    => 'POST',
+      'cookie'    => 'UIDADMIN=' + datastore['UIDADMIN']
```

Comment:
I notice you've removed the `UIDADMIN` option from the module `datastore`, and replaced it wtih `@session_cookie`. Should `@session_cookie` also be used here?

Also, see my other review comments about `UIDADMIN`.

Reply:
Yes, `@session_cookie` should have been used here. I've made that consistent across the module as of commit 29627331cfeeb35d4a74d5128665e7e5d0f6d514.